### ISSUE TITLE
The register data cannot be viewed.

### ISSUE THREAD

**Issue by ghost** [JesusFreke/smalidea#16]
![image](https://user-images.githubusercontent.com/27600008/110487584-af9e6380-8128-11eb-8b1e-34b94f4de207.png)
![image](https://user-images.githubusercontent.com/27600008/110487787-e1172f00-8128-11eb-9d97-c63397d370d9.png)
smalidea-0.0.6

**Comment by JesusFreke** [comment]
Please provide any logs or errors. See, e.g. Help -> show log in Files.

**Comment by NikitinWork** [comment]
![image](https://user-images.githubusercontent.com/43878577/111437193-df92cb80-8713-11eb-8ccc-91cedb6036e8.png)
[idea.log](https://github.com/JesusFreke/smalidea/files/6154810/idea.log)

**Comment by JesusFreke** [comment]
The log doesn't have any useful information, unfortunately. And your screenshot doesn't give much context.

Without more information that may point me to what the problem might be, I can't really do anything unless I'm able to reproduce the problem.

**Comment by NikitinWork** [comment]
I can provide any information. I can also show you everything through discord. If it helps you with the solution of the problems

**Comment by JesusFreke** [comment]
The full method being debugged would be useful. And the execution point at which you are not able to see the registers.

**Comment by NikitinWork** [comment]
@JesusFreke Can't get registers anywhere. That being said, everything else works perfectly.

**Comment by JesusFreke** [comment]
Oh, I forgot the most basic thing. Which version of Android is your device running? The last few versions had a bug in art that prevented this from working. But it was fixed in Android 11 (R). I think it was P and Q that had the issue.

**Comment by NikitinWork** [comment]
I have to use very bad and outdated devices on android 7.0.
Registers work on version 0.0.5

Additional information just in case.
Using Magisk-v20.4
ro.rock.gota.version: [MMX_Q402+_RU_SW_V15_HW_V4.0_20180906
ro.product.cpu.abilist: [armeabi-v7a,armeabi]
gsm.version.baseband: [FM_BASE_15C_W17.15.4_P17|sc983x_modem|12-06-2017 19:08:33]

UPD:
Realme C11, android 10
Magisk-v22.1

Only version 0.05 works

**Comment by detunized** [comment]
I confirm I have the same. I cannot view any of the variables `v*` or `p*`. I tried on an emulator and on two OnePlus devices. One running Android 10 and the other running Android 7.1.1. Could it be related to this change?
https://github.com/JesusFreke/smalidea/commit/7d29272dc036c96a69577d34104ab9967e437a9e

**Comment by mxl9236** [comment]
First of all, I am very grateful to the author for updating this plugin. 
This problem seems to be related to the version of android studio. 
This problem occurs when I use android studio 4.1.x, and it works normally when I change to android studio 4.2.x.

**Comment by detunized** [comment]
@mxl9236, thanks for the tip. I will try to upgrade. I was on the latest at the moment of reporting this problem.

**Comment by gongwuchun** [comment]
> Oh, I forgot the most basic thing. Which version of Android is your device running? The last few versions had a bug in art that prevented this from working. But it was fixed in Android 11 (R). I think it was P and Q that had the issue.

Yes, It works on Android R but Failed on Android P, Can I backport the fix patch from Android R to Android P? Thanks!

**Comment by zlwen** [comment]
I used Codex to port the original code to the new IntelliJ SDK. Viewing register values is working now—you can test it out!
https://github.com/zlwen/smalidea/tree/upgrade-intellij-platform-2x

**Comment by zlwen** [comment]
or use the prebuilt binary https://github.com/zlwen/smalidea/releases/download/0.0.7/smalidea-0.0.7-dev.zip
